crates/storage/provider/src/transaction.rs

```diff
@@ -760,3 +1474,68 @@ pub enum TransactionError {
         block_hash: BlockHash,
     },
 }
+
+#[cfg(test)]
+mod test {
+    use crate::{insert_canonical_block, test_utils::blocks::*, Transaction};
+    use reth_db::{mdbx::test_utils::create_test_rw_db, tables, transaction::DbTxMut};
+    use reth_primitives::{proofs::EMPTY_ROOT, ChainSpecBuilder, MAINNET};
+    use std::ops::DerefMut;
+
+    #[test]
+    fn insert_get_take() {
```

Comment:
There are way too few tests here considering everything in the stages crate has been moved here at this point.

Reply:
What is moved from stage still gets tested from it. With test data i tried to cover cases that can happen in both getting and taking data from database, and additionally, BlockchainTree is testing this functionality in its own way.

Reply:
As part of the refactor of the Transaction struct during our tech debt cleanup, we should introduce more tests